lib/ipmi_fru.c

```diff
@@ -832,11 +845,12 @@ read_fru_area_section(struct ipmi_intf * intf, struct fru_info *fru, uint8_t id,
 			break;
 		}
 		if (rsp->ccode) {
-			/* if we get C7 or C8  or CA return code then we requested too
-			* many bytes at once so try again with smaller size */
-			if ((rsp->ccode == 0xc7 || rsp->ccode == 0xc8 || rsp->ccode == 0xca) &&
-				(--fru_data_rqst_size > 8)) {
-				lprintf(LOG_INFO, "Retrying FRU read with request size %d",
+			/* if we get C7 or C8  or CA return code then we
+			 * requested too many bytes at once so try again with
+			 * smaller size */
+			if (fru_cc_rq2big(rsp->ccode) && (--fru_data_rqst_size > 8)) {
```

Comment:
Why do we still have `8` here?
Isn't this the same case as in other functions that check for >32 and reduce by 8 ?

Reply:
I have been jetlagged since I got back from Christmas, if you mean, let's just swap the number 8 here for the macro because it's the same value it's using, then yeah, let's do that -- just missed this one.  If you think the logic is wrong here, then that's a different story.